src/homeassistant.cpp

```diff
@@ -58,6 +59,14 @@ HomeAssistant::HomeAssistant(const QVariantMap &config, EntitiesInterface *entit
             QVariantMap map = iter.value().toMap();
             m_ip            = map.value(Integration::KEY_DATA_IP).toString();
             m_token         = map.value(Integration::KEY_DATA_TOKEN).toString();
+            //            m_ssl           = map.value(Integration::KEY_DATA_SSL).toBool();
+            m_ssl = map.value("ssl").toBool();
+
+            if (m_ssl) {
+                m_url = QString("wss://").append(m_ip).append("/api/websocket");
+            } else {
+                m_url = QString("ws://").append(m_ip).append("/api/websocket");
+            }
```

Comment:
Simplify, since SSL only affects the prefix:
`m_url = QString(m_ssl ? "wss://" : "ws://").append(m_ip).append("/api/websocket");`